src/NBitcoin/ChainedBlock.cs

```diff
@@ -267,10 +312,11 @@ public Target GetWorkRequired(Consensus consensus)
             }
 
             ChainedBlock firstChainedBlock = this.EnumerateToGenesis().FirstOrDefault(o => o.Height == pastHeight);
-            assert(firstChainedBlock);
+            if (firstChainedBlock == null)
+                throw new NotSupportedException("Can only calculate work of a full chain");
```

Comment:
Also add a Guard.NotNull here

Reply:
this will be for later when we have those guards, so non issue here